google/cloud/spanner_dbapi/connection.py

```diff
@@ -141,14 +144,23 @@ def inside_transaction(self):
         """Flag: transaction is started.
 
         Returns:
-            bool: True if transaction begun, False otherwise.
+            bool: True if transaction started, False otherwise.
         """
         return (
             self._transaction
             and not self._transaction.committed
             and not self._transaction.rolled_back
         )
 
+    @property
+    def transaction_begun(self):
+        """Flag: transaction has begun
+
+        Returns:
+            bool: True if transaction begun, False otherwise.
+        """
+        return (not self._autocommit) or self._transaction_begin_marked
```

Comment:
The Java Connection API has three states for this (if I remember correctly, it uses two boolean flags to monitor it):
1. `isInTransaction`: This returns true if `(not self._autocommit) or self._transaction_begin_marked`.
2. `isTransactionStarted`: This returns true if `isInTransaction() && at_least_one_statement_has_been_executed`

The difference here is the following:
1. `isInTransaction` indicates that the connection has a transaction (i.e. will not use auto-commit).
2. `isTransactionStarted` indicates that the transaction has executed at least one statement. This is also equivalent to 'Cloud Spanner knows about the transaction, and the connection has a transaction ID'.

One of the following client-side statements that we will be implementing is `SET TRANSACTION READ ONLY`. This statement is __only allowed__ when `isInTransaction() == true && isTransactionStarted() == false`. The reason for this is that:
1. Setting a property of a transaction, such as whether it should be a read-only or a read/write transaction, is something that you can only do before you actually start the transaction.
2. Setting a property of a transaction when you don't have a transaction does not make sense.

I think it would be good to implement both the above-mentioned methods in this pull request. That guarantees that we don't change the behavior of any of those methods when implementing any following client-side statements.

Reply:
We have a property inside_transaction() in Connection which is same as isTransactionStarted. Can you please check that?